contracts/metagov/BalancerGaugeStaker.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "../pcv/PCVDeposit.sol";
+import "./utils/LiquidityGaugeManager.sol";
+import "../external/balancer/IBalancerMinter.sol";
+
+/// @title Deposit that can stake in Balancer gauges
+/// @author Fei Protocol
+contract BalancerGaugeStaker is PCVDeposit, LiquidityGaugeManager {
+    address public constant BALANCER_GAUGE_CONTROLLER = 0xC128468b7Ce63eA702C1f104D55A2566b13D3ABD;
+    address public constant BALANCER_MINTER = 0x239e55F427D44C3cc793f49bFB507ebe76638a2b;
+    address public constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+
+    /// @notice Balancer gauge staker
+    /// @param _core Fei Core for reference
+    constructor(address _core) CoreRef(_core) LiquidityGaugeManager(BALANCER_GAUGE_CONTROLLER) {}
+
+    /// @notice returns total balance of PCV in the Deposit
+    function balance() public view override returns (uint256) {
+        return IERC20(BAL).balanceOf(address(this));
+    }
+
+    /// @notice gets the token address in which this deposit returns its balance
+    function balanceReportedIn() public view virtual override returns (address) {
+        return BAL;
+    }
+
+    /// @notice gets the resistant token balance and protocol owned fei of this deposit
+    function resistantBalanceAndFei() public view virtual override returns (uint256, uint256) {
+        return (balance(), 0);
+    }
+
+    /// @notice noop
+    function deposit() external override {}
+
+    /// @notice withdraw BAL held to another address
+    /// the BAL rewards accrue on this PCVDeposit when Gauge rewards are claimed.
+    function withdraw(address to, uint256 amount) external override onlyPCVController whenNotPaused {
+        IERC20(BAL).transfer(to, amount);
+    }
+
+    /// @notice withdraw ERC20 from the contract
+    /// @param token address of the ERC20 to send
+    /// @param to address destination of the ERC20
+    /// @param amount quantity of ERC20 to send
+    function withdrawERC20(
+        address token,
+        address to,
+        uint256 amount
+    ) public virtual override onlyPCVController {
+        // if the token has a gauge, we unstake it from the gauge before transfer
+        // @dev : note that this gauge unstaking normally is access controlled to
+        // onlyTribeRole(TribeRoles.METAGOVERNANCE_GAUGE_ADMIN), but since
+        // withdrawERC20() is a higher clearance role (PCVController), this
```

Comment:
Is there any situation where we will want to unstake from the gauge but not immediately withdraw the tokens? Is it worthwhile creating a `function unstakeFromGauge(address token, uint256 amount) hasAnyOfTwoRoles(TribeRoles.METAGOVERNANCE_GAUGE_ADMIN, TribeRoles.PCV_CONTROLLER) public` method?

Reply:
this is equivalent to withdrawERC20 with the contract itself as the target address, so I don't think we need another method